src/main/java/net/sf/mzmine/desktop/impl/MainWindow.java

```diff
@@ -90,48 +92,76 @@ public HelpImpl getHelpImpl() {
   /**
    * WindowListener interface implementation
    */
+  @Override
   public void windowOpened(WindowEvent e) {}
 
+  @Override
   public void windowClosing(WindowEvent e) {
     exitMZmine();
   }
 
+  @Override
   public void windowClosed(WindowEvent e) {}
 
+  @Override
   public void windowIconified(WindowEvent e) {}
 
+  @Override
   public void windowDeiconified(WindowEvent e) {}
 
+  @Override
   public void windowActivated(WindowEvent e) {}
 
+  @Override
   public void windowDeactivated(WindowEvent e) {}
 
+  @Override
   public void setStatusBarText(String text) {
     setStatusBarText(text, Color.black);
   }
 
   /**
    */
+  @Override
   public void displayMessage(Window window, String msg) {
     displayMessage(window, "Message", msg, JOptionPane.INFORMATION_MESSAGE);
   }
 
   /**
    */
+  @Override
   public void displayMessage(Window window, String title, String msg) {
     displayMessage(window, title, msg, JOptionPane.INFORMATION_MESSAGE);
   }
 
+  @Override
   public void displayErrorMessage(Window window, String msg) {
     displayMessage(window, "Error", msg);
   }
 
+  @Override
   public void displayErrorMessage(Window window, String title, String msg) {
     displayMessage(window, title, msg, JOptionPane.ERROR_MESSAGE);
   }
 
   public void displayMessage(Window window, String title, String msg, int type) {
 
+    // sending error message with a maximum of 5
+    if (MZminePreferences.sendErrorEMail.getValue() != null
+        && MZminePreferences.sendErrorEMail.getValue() && mailCounter < 5) {
+      ErrorMail errorMail = new ErrorMail();
+      try {
+        errorMail.sendErrorEmail(ErrorMailSettings.eMailAddress.getValue(),

```

Comment:
I think you should send the email in a separate thread, because the current implementation will block the GUI thread while the email is being sent. Perhaps the best is to make ErrorMail implement Runnable, then create a new Thread with it.